src/prefs/TelemetryPrefs.cpp

```diff
@@ -0,0 +1,113 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  TelemetryPrefs.cpp
+
+  Dmitry Vedenko
+
+
+*******************************************************************//**
+
+\class TelemetryPrefs
+\brief A PrefsPanel to enable/disable user data sharing.
+
+*//*******************************************************************/
+
+#include "../Audacity.h"
+#include "TelemetryPrefs.h"
+
+#include <wx/defs.h>
+#include <wx/hyperlink.h>
+
+#include "../ShuttleGui.h"
+
+////////////////////////////////////////////////////////////////////////////////
+
+TelemetryPrefs::TelemetryPrefs(wxWindow * parent, wxWindowID winid)
+:  PrefsPanel(parent, winid, XO("Analytics"))
+{
+   Populate();
+}
+
+TelemetryPrefs::~TelemetryPrefs()
+{
+}
+
+ComponentInterfaceSymbol TelemetryPrefs::GetSymbol()
+{
+   return TELEMETRY_PREFS_PLUGIN_SYMBOL;
+}
+
+TranslatableString TelemetryPrefs::GetDescription()
+{
+   return XO("Preferences for Analytics");
+}
+
+wxString TelemetryPrefs::HelpPageName()
+{
+   return "Analytics_Preferences";
+}
+
+void TelemetryPrefs::Populate()
+{
+   //------------------------- Main section --------------------
+   // Now construct the GUI itself.
+   // Use 'eIsCreatingFromPrefs' so that the GUI is
+   // initialized with values from gPrefs.
+   ShuttleGui S(this, eIsCreatingFromPrefs);
+   PopulateOrExchange(S);
+   // ----------------------- End of main section --------------
+}
+
+void TelemetryPrefs::PopulateOrExchange(ShuttleGui & S)
+{
+   S.SetBorder(2);
+
+   S.StartStatic(XO("Analytics"));
+   {
+       S.StartHorizontalLay (wxEXPAND);
+       {
+           S.StartVerticalLay (0);
+           {
+               S.TieCheckBox ({}, { wxT ("/Telemetry/Enabled"), false });
+           }
+           S.EndVerticalLay ();
+
+           S.StartVerticalLay ();
+           {
+               S.AddFixedText (XO ("Share anonymous analytics and crash reports"));
+               S.AddFixedText (XO ("Help Audacity improve its service, by allowing analytics of usage and data."));
+               S.AddWindow (
+                   safenew wxHyperlinkCtrl (
+                       S.GetParent (), wxID_ANY, 
+                       XO ("Analytics Privacy Policy").Translation(), 
+                       "https://www.audacityteam.org/contact/privacy-policy/"),
```

Comment:
This URL does not currently show a privacy policy. That should be online *before* merging this.

Moreover, a wall of legalese is not really sufficient for informed consent. There should be concise summaries of what information is being collected and why directly in the GUI, not only on a webpage external to the application.

Reply:
Indeed. The privacy policy will probably need, among other things, to list Yandex and Google as sub-processors, state that the user's personal data will be sent outside the EU, and probably offer a DPO contact, in addition to the usual things.

You can find instructions at
https://ec.europa.eu/info/law/law-topic/data-protection/reform/rules-business-and-organisations_en
https://edps.europa.eu/data-protection/eu-institutions-dpo/case-law-guidance_en

Spoiler: it's going to be practically impossible to respect GDPR and Schrems II if you use Google.

Reply:
Surely you need to include the privacy policy in the distributed software, rather than asking the user to view it? Surely when I download a copy of Audacity, I didn't agree to a copy of privacy policy that you can update at any time.